### PR TITLE
Add recent GHCs to CI config, switch to cryptohash-md5

### PR DESCRIPTION
**Pull request by jhrcek**
- I regenerated CI config to include 3 most recent GHC versions
- I remove GHC 8.4.4 from CI as it seems unnecessary to test (it seems to me that supporting last 6 major versions of GHC is more then sufficient). Also GHC 8.4.4 setup using ghcup is broken (would have to use alternative installation method to make it work)
- I addressed this issue: https://github.com/andrevdm/postgresql-migration/issues/15
- I removed all unused / duplicated stuff from build-depends in cabal file.

### CODE REVIEW

File: postgresql-migration.cabal [hunk 1]
```diff
@@ -36,7 +41,9 @@ common common-ghc-options-ide
   ghc-options: -fwrite-ide-info -hiedir=.hie
 
 common common-ghc-options
-  ghc-options: -Wall -Wincomplete-uni-patterns -Wincomplete-record-updates -Wcompat -Wredundant-constraints -Wnoncanonical-monad-instances -Widentities -fhide-source-paths -Wpartial-fields -fhide-source-paths -freverse-errors
```
Comment (jhrcek) [inline postgresql-migration.cabal:39]: Just removed redundant `-fhide-source-paths` which was in the list 2x

**Review by andrevdm — APPROVED** [review]

**Comment by andrevdm** [comment]
Thanks @jhrcek.

I've just merged and am about to test

**Comment by andrevdm** [comment]
There were a couple of merge issues. My fault as I was also making some bounds changes :facepalm:. But it looks like all your changes are there and seem to be working fine. Thanks again

**Comment by andrevdm** [comment]
Pushed to hackage in 0.2.1.7. I decided on a version bump rather than just a metadata revision for this one

**Pull request merged**